src/PBot/Issues/InvolvedIssueQuery.cs

```diff
@@ -0,0 +1,81 @@
+namespace PBot.Issues
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using System.Threading.Tasks;
+    using Octokit;
+
+    internal class InvolvedIssueQuery
+    {
+        private static readonly Regex TaskForceRegex = new Regex(@"Task[\s-]?Force:\s*(.*)$", RegexOptions.Multiline | RegexOptions.IgnoreCase | RegexOptions.Compiled);
+        private static readonly Regex MentionRegex = new Regex(@"@[a-z0-9.-]+", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+        private readonly GitHubClient client;
+
+        public InvolvedIssueQuery(GitHubClient client)
+        {
+            this.client = client;
+        }
+
+        public async Task<IEnumerable<InvolvedIssue>> Perform(string username)
+        {
+            var repos = (await client.Repository.GetAllForOrg("Particular")).ToList();
+            var tasks = repos.Select(x => GetInvolvedIssuesForRepo(x, username));
+
+            return (await Task.WhenAll(tasks)).SelectMany(issue => issue);
+        }
+
+        private async Task<IEnumerable<InvolvedIssue>> GetInvolvedIssuesForRepo(Repository repo, string username)
+        {
+            var mentionedFilter = new RepositoryIssueRequest { State = ItemState.Open, Mentioned = username };
+            var assigneeFilter = new RepositoryIssueRequest { State = ItemState.Open, Assignee = username };
+            var creatorFilter = new RepositoryIssueRequest { State = ItemState.Open, Creator = username };
+
+            var issues = (await client.Issue.GetForRepository("Particular", repo.Name, mentionedFilter))
+                .Concat(await client.Issue.GetForRepository("Particular", repo.Name, assigneeFilter))
+                .Concat(await client.Issue.GetForRepository("Particular", repo.Name, creatorFilter))
```

Comment:
Yep, that's true. The `await` will free up the thread, but the queries will happen serially.

Note that this is already the case in the existing `InvolvedIssueQuery.cs` (git hasn't recognised this file as a rename). I've just added a third query to the two existing ones.

Anyhow, I'll see if I can add a commit to parallelise them.